### ISSUE TITLE
The editor doesn't preserve text alignment when changing to a Hn or paragraph node.

### ISSUE THREAD

**Issue by Haja-dev** [ueberdosis/tiptap#3644]
### What’s the bug you are facing?

I have a text that has text alignment for example align center( or right, justify).
![image](https://user-images.githubusercontent.com/54202486/213990668-948b517d-bc21-42bb-b614-635506800e13.png)

When I change the text to a Hn node, for example to h2, the alignment I had before is not preserved. 
![image](https://user-images.githubusercontent.com/54202486/213991171-9d1b0e47-8af7-43dc-ae62-59c8243316cd.png)

### Which browser was this experienced in? Are any special extensions installed?

All

### How can we reproduce the bug on our side?

1. Install Tiptap with Starter Kit and add TextAlign extension
2. Add a text in the editor and change the text alignment to another one than left
3. Then, change the title type to h1 or another Hn or a paragraph.

### What did you expect to happen?

When changing text to Hn or paragraph node, the text alignment should be preserved.

### Did you update your dependencies?

- [X] Yes, I’ve updated my dependencies to use the latest version of all packages.

### Are you sponsoring us?

- [ ] Yes, I’m a sponsor. 💖

**Comment by hkurra** [comment]
@Haja-dev Quick solution for Headings: Just pass other attributes along with level 

`
 let otherAttributes = this.editor.getAttributes('heading')
if (isEmpty(otherAttributes)) {
    otherAttributes = this.editor.getAttributes('paragraph')
}
editor.chain().focus().toggleHeading({...otherAttributes, level: Number(newLevel) as Level})
`
the issue will still happen with keyboard shortcut so to avoid that you can extend the default heading node and override the keyboard shortcut 

`
addKeyboardShortcuts() {
    return this.options.levels.reduce(
      (items, level) => ({
        ...items,
        ...{
          ['Mod-Alt-${level}']: () => {
            let otherAttributes = this.editor.getAttributes('heading')
            if (isEmpty(otherAttributes)) {
              otherAttributes = this.editor.getAttributes('paragraph')
            }
            this.editor.commands.toggleHeading({ ...otherAttributes, level })
          }
        },
      }),
      {}
    )
  },
  `

**Comment by seancrater** [comment]
Encountered this same issue when switching from Paragraph Nodes to List Nodes

**Comment by harrisonhoward** [comment]
Any update on fixing this in the extension itself? This occurs when switching to paragraphs if there is a hack for that.

**Comment by nperez0111** [comment]
toggleHeading could be implemented with something like this:

```
      toggleHeading: attributes => ({ commands, state }) => {
        if (!this.options.levels.includes(attributes.level)) {
          return false
        }
        Object.assign(attributes, state.selection.$anchor.parent.attrs)

        return commands.toggleNode(this.name, 'paragraph', attributes)
      },
 ```
 
 But, it is brittle because it only considers the $anchor of the selection, when the selection can be over a range (i.e. it can copy over the wrong attributes). It is a bit unclear what the expected behavior should be for converting one node type to another which is why we don't currently copy over the attributes. I guess, in some situations it may be obvious; for example if the selection is collapsed (to & from are equal) then it is a 1-1 mapping and we should probably copy over the attributes.
 

Well, in writing this comment I ended up writing some sort of a solution to it (got nerd sniped). I have a PR here for it: https://github.com/ueberdosis/tiptap/pull/5489

**Comment by nperez0111** [comment]
This has been released with 2.7.0-pre.0 and will be released in tiptap 2.7.0 when this beta has been tested by more users

**Comment by nperez0111** [comment]
This is released with v2.7.0 https://github.com/ueberdosis/tiptap/releases/tag/v2.7.0
